[DOCS-13601] Add SDS best practices section (#36874)
DataDog/documentation: content/en/observability_pipelines/processors/sensitive_data_scanner.md

You can set up the pipeline and processor in the [UI](#set-up-the-processor-in-the-ui), [API][10], or [Terraform](#set-up-the-processor-using-terraform).

> See [Best practices to optimize performance](#best-practices-to-optimize-performance) for configuration recommendations.

Reviewer: ```suggestion
See [Best practices to optimize performance](#best-practices-to-optimize-performance) for tips on reducing resource usage.
```

1. Select your pipeline.
1. Click the Sensitive Data Scanner processor to expand it.
1. Click **View Scanning Rules** to open the side panel and see **Matches in the last 24 hours** for each rule.

See [Delete a rule](#delete-a-rule) if you want to delete an unused rule.

### Only scan the events and fields that need to be scanned for sensitive data

If you know the types of events you want to scan, define a processor query that only sends the events you want to the processor.

Reduce scanning time by targeting specific event attributes for scanning or excluding event attributes from being scanned. See the **Define rule target and conditions** section in [Set up the processor](#set-up-the-processor-in-the-ui).

### Evaluate and benchmark performance optimizations

Use the `pipelines.component_latency_seconds` metric to:

- Benchmark processor performance when you add a rule
- Evaluate performance after making optimization changes, such as reducing the number of fields being scanned and removing unused rules

To view the `pipelines.component_latency_seconds` metric:

1. Navigate to [Metrics Explorer][11].
1. In the metric field, enter `pipelines.component_latency_seconds`.

> 1. In the **from** field, enter the tags: `component_type:sensitive_data_scanner` and `component_id:<COMPONENT_ID>`, where `<COMPONENT_ID>` is the ID for your Sensitive Data Scanner processor.

Author: ```suggestion
1. In the **from** field, enter the tag `component_id:<COMPONENT_ID>`, where `<COMPONENT_ID>` is the ID for your Sensitive Data Scanner processor.
```

}
{{< /code-block >}}

## Best practices to optimize performance

> The Sensitive Data Scanner processor is CPU intensive, which means more resources are needed when there are more fields and data to scan. The following are best practices to optimize performance.

Reviewer: Minor suggestion to tighten.
```suggestion
The Sensitive Data Scanner processor is CPU intensive. Use the following best practices to optimize performance.
```

### Only enable rules you need

Rules that are enabled but not used consume unnecessary resources. Check the Sensitive Data Scanner processor to view how many matches each rule has had over the past 24 hours.

1. Navigate to [Observability Pipelines][2].
1. Select your pipeline.
1. Click the Sensitive Data Scanner processor to expand it.
1. Click **View Scanning Rules** to open the side panel and see **Matches in the last 24 hours** for each rule.

> See [Delete a rule](#delete-a-rule) if you want to delete an unused rule.

Reviewer: ```suggestion
See [Delete a rule](#delete-a-rule) to delete an unused rule.
```
